ts/components/BottomSheetComponent.tsx

```diff
@@ -0,0 +1,182 @@
+
+import I18n from "../i18n";
+import customVariables from "../theme/variables";
+import { isScreenReaderEnabled } from "../utils/accessibility";
+import ButtonDefaultOpacity from "./ButtonDefaultOpacity";
+import { IOColors } from "./core/variables/IOColors";
+import { IOStyles } from "./core/variables/IOStyles";
+import AppHeader from "./ui/AppHeader";
+import IconFont from "./ui/IconFont";
+
+type Props = {
+  bottomSheetTitle: string;
+  handleClose: () => void;
+  maxSnapPoint: number;
+  content?: React.ReactNode;
+};
+
+const styles = StyleSheet.create({
+  row: {
+    flexDirection: "row",
+    alignItems: "center",
+    justifyContent: "space-between"
+  },
+  title: {
+    fontSize: 18,
+    color: customVariables.lightGray,
+    alignSelf: "center",
+    flex: 1,
+    flexDirection: "row",
+    justifyContent: "flex-start",
+    lineHeight: customVariables.lineHeightBase
+  },
+  modalClose: {
+    flex: 1,
+    paddingRight: 0,
+    flexDirection: "row",
+    justifyContent: "flex-end"
+  },
+  icon: {
+    paddingRight: 0
+  },
+  container: {
+    alignSelf: "center",
+    width: "100%",
+    borderTopRightRadius: 16,
+    borderTopLeftRadius: 16,
+    backgroundColor: customVariables.colorWhite,
+    ...IOStyles.horizontalContentPadding,
+    paddingTop: 16
+  },
+  content: {
+    ...IOStyles.horizontalContentPadding,
+    width: "100%",
+    height: "100%",
+    backgroundColor: IOColors.white
+  },
+  hover: {
+    minWidth: "100%",
+    minHeight: "100%",
+    bottom: 0,
+    left: 0,
+    top: 0,
+    position: "absolute",
+    alignItems: "center",
+    justifyContent: "center"
+  }
+});
+
+const BottomSheetComponent: React.FunctionComponent<Props> = (props: Props) => {
+  const bottomSheetRef = React.useRef<BottomSheet>(null);
+  const [opacity] = React.useState(new Animated.Value(0));
+  const [accessible, setAccessible] = React.useState(false);
+
+  const color = opacity.interpolate({
+    inputRange: [0, 1],
+    outputRange: ["rgba(0,0,0,0)", "rgba(0,0,0,0.5)"]
+  });
+
+  const modalBackdrop = {
+    backgroundColor: color
+  };
+
+  const openBackdropAnimation = () =>
+    Animated.timing(opacity, {
+      useNativeDriver: false,
+      toValue: 1,
+      duration: 200
+    }).start();
+
+  const closeBackdropAnimation = () =>
+    Animated.timing(opacity, {
+      useNativeDriver: false,
+      toValue: 0,
+      duration: 200
+    }).start();
+
+  React.useEffect(() => {
+    if (props.content) {
+      openBottomSheet().catch(_ => closeBottomSheet());
+    } else {
+      closeBottomSheet();
+    }
+  }, [props.content]);
+
+  useHardwareBackButton(() => {
+    closeBottomSheet();
+    return true;
+  });
+
+  const closeBottomSheet = () => {
+    setAccessible(false);
+    if (bottomSheetRef && bottomSheetRef.current) {
+      bottomSheetRef.current.snapTo(0);
+      closeBackdropAnimation();
+      props.handleClose();
+    }
+  };
```

Comment:
```suggestion
  const closeBottomSheet = () => {
    setAccessible(false);
    if (bottomSheetRef && bottomSheetRef.current) {
      bottomSheetRef.current.snapTo(0);
    }
  };
```
When the bottomsheet is closed with the `X` there is a slightly different graphical behavior.
It doesn't do the exit animation and when it is called next time it does not do the entry animation.
<img src=https://user-images.githubusercontent.com/26501317/96766272-1cfc6280-13db-11eb-829d-9abe98320467.gif width=300 />

Removing these lines works for me, because the backdropanimation is executed by the callback, but I don't know if removing ` props.handleClose();` generate any others side effects.